In [5]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt
import seaborn as sns

import os
print(os.getcwd()) 



C:\Users\matze\Documents\Winton\CRISPDM


In [7]:
X = pd.read_csv('X_features.csv', index_col=0)
y = pd.read_csv('y_features.csv', index_col=0)

### Score and Benchmark

In [10]:
from sklearn.metrics import mean_absolute_error
zeros = pd.DataFrame(np.zeros(y.shape), columns = y.columns)


In [13]:
test_score = mean_absolute_error(y, zeros)
test_score_minutes = mean_absolute_error(y.loc[:,'Ret_121':'Ret_180'], zeros.loc[:,'Ret_121':'Ret_180'])
test_score_daily = mean_absolute_error(y.loc[:,'Ret_PlusOne':'Ret_PlusTwo'], zeros.loc[:,'Ret_PlusOne':'Ret_PlusTwo'])
print(test_score)
print(test_score_minutes)
print(test_score_daily)

0.021625530418345765
0.0006331533763565462
0.0155220073711832


## Linear Regression

In [2]:
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
linreg = LinearRegression()
ridge = Ridge()
lasso = Lasso()

In [101]:
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import GroupKFold

winton_cv = cross_val_score(linreg, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00061921 -0.00062317 -0.00066605 -0.000601   -0.00059505 -0.00073446
 -0.00062817 -0.00066855 -0.00059782 -0.00055231]
Score Mean: -0.0006285793252795579
Score Max: -0.000552307075398458 Score Min: -0.0007344601337599301
Benchmark: 0.0006331533763565462


In [106]:
ridge = Ridge(normalize=True)
winton_cv = cross_val_score(ridge, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00059576 -0.00060948 -0.00065632 -0.00058282 -0.00058428 -0.00074498
 -0.00062312 -0.0006665  -0.00058426 -0.00053056]
Score Mean: -0.0006178083596747471
Score Max: -0.0005305583771803281 Score Min: -0.000744977287942406
Benchmark: 0.0006331533763565462


In [19]:
winton_cv = cross_val_score(lasso, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00058496 -0.0005963  -0.00064945 -0.00056168 -0.00057212 -0.00074801
 -0.00060083 -0.00067214 -0.0005735  -0.00052119]
Score Mean: -0.0006080174419997037
Score Max: -0.0005211943114080878 Score Min: -0.0007480140423706209
Benchmark: 0.0006331533763565462


In [76]:
lasso= Lasso(normalize =True)
winton_cv = cross_val_score(lasso, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00058496 -0.00059649 -0.00064963 -0.00056113 -0.00057216 -0.00074801
 -0.00060069 -0.00067211 -0.0005735  -0.00052046]
Score Mean: -0.0006079147761024154
Score Max: -0.0005204633439076678 Score Min: -0.0007480140423706209
Benchmark: 0.0006331533763565462


In [108]:
winton_cv = cross_val_score(lasso, X, y.Ret_Next_10, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))
mean_absolute_error(y.Ret_Next_10, zeros.Ret_121)


Score: [-0.00149765 -0.00187759 -0.00174793 -0.00183153 -0.00182986 -0.00196353
 -0.00175625 -0.00206738 -0.00158528 -0.00185007]
Score Mean: -0.0018007064235862566
Score Max: -0.0014976471264876677 Score Min: -0.00206738165804562
Benchmark: 0.0006331533763565462


0.001797113925478407

In [110]:
winton_cv = cross_val_score(lasso, X, y.Ret_PlusOne, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_daily))



Score: [-0.01396033 -0.01483157 -0.01694395 -0.01532647 -0.01487931 -0.01590525
 -0.01594459 -0.01826528 -0.01510128 -0.016546  ]
Score Mean: -0.01577040256202347
Score Max: -0.013960331817667208 Score Min: -0.01826527618004748
Benchmark: 0.0155220073711832


lasso mit X_Interaction:
Score: [-0.00058796 -0.00059969 -0.00064973 -0.00057151 -0.00057197 -0.00074997
 -0.00060218 -0.00067318 -0.00056749 -0.000529  ]
Score Mean: -0.0006102710573105517
Score Max: -0.0005290048458109724 Score Min: -0.0007499733011142033
Benchmark: 0.0155220073711832

### Random Forets

In [111]:
from sklearn.ensemble import RandomForestRegressor
regr1 = RandomForestRegressor(n_estimators =100 ,random_state=0)
winton_cv = cross_val_score(regr1, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))


Score: [-0.00059818 -0.00064709 -0.00065108 -0.00058882 -0.00061449 -0.0007332
 -0.00062632 -0.00074564 -0.00061879 -0.00054931]
Score Mean: -0.0006372905437952748
Score Max: -0.0005493056638844501 Score Min: -0.0007456384552396402
Benchmark: 0.0006331533763565462


In [ ]:
regr2 = RandomForestRegressor(n_estimators =100, criterion= 'mse', max_depth= 8, min_samples_leaf= 20, min_samples_split= 10)
winton_cv = cross_val_score(regr2, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

### GBR

In [23]:
from sklearn.ensemble import GradientBoostingRegressor
gbr1 = GradientBoostingRegressor(n_estimators =100)
winton_cv = cross_val_score(gbr1, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00058766 -0.00062389 -0.00064821 -0.00057316 -0.00057348 -0.00071936
 -0.00060596 -0.00067042 -0.00059935 -0.0005306 ]
Score Mean: -0.0006132104333579254
Score Max: -0.0005306000757901328 Score Min: -0.000719362571550964
Benchmark: 0.0006331533763565462


In [27]:
from sklearn.ensemble import GradientBoostingRegressor
gbr1 = GradientBoostingRegressor(n_estimators =100)
winton_cv = cross_val_score(gbr1, X, y.Ret_122, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00056574 -0.00070529 -0.00078044 -0.00064104 -0.0006669  -0.00072632
 -0.00063581 -0.00078307 -0.0006907  -0.00061852]
Score Mean: -0.0006813843941282283
Score Max: -0.0005657441357384968 Score Min: -0.0007830691060645135
Benchmark: 0.0006331533763565462


In [29]:
gbr1 = GradientBoostingRegressor(n_estimators =30)
winton_cv = cross_val_score(gbr1, X, y.Ret_121, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_minutes))

Score: [-0.00057902 -0.0006058  -0.00064332 -0.00056648 -0.00056851 -0.00071577
 -0.0006062  -0.00066789 -0.00059362 -0.00052623]
Score Mean: -0.0006072831737799556
Score Max: -0.0005262345068230863 Score Min: -0.0007157664404122108
Benchmark: 0.0006331533763565462


In [40]:
from sklearn.ensemble import GradientBoostingClassifier
gbr1 = GradientBoostingRegressor(n_estimators =30)
winton_cv = cross_val_score(gbr1, X, y.MAD_Next_5, X.Feature_7, cv=GroupKFold(n_splits=10),scoring='neg_mean_absolute_error')
print('Score: {}'.format(winton_cv))
print('Score Mean: {}'.format(winton_cv.mean()))
print('Score Max: {}'.format(winton_cv.max()), 'Score Min: {}'.format(winton_cv.min()))
print('Benchmark: {}'.format(test_score_daily))
mean_absolute_error(y.Ret_Next_5, zeros.Ret_121)

Score: [-0.00021034 -0.00021172 -0.00024116 -0.00021874 -0.00021826 -0.00025662
 -0.00023138 -0.00026151 -0.00024776 -0.00021152]
Score Mean: -0.00023090284487026227
Score Max: -0.00021034073747926773 Score Min: -0.00026151366545320876
Benchmark: 0.0155220073711832


0.0012023219862151305

In [54]:

#np.repeat(y.MAD_Next_5.mean(),y.MAD_Next_5.size)
#y.MAD_Next_5
mean_absolute_error(y.MAD_Next_5 ,pd.DataFrame(np.repeat(y.MAD_Next_5.mean(),y.MAD_Next_5.size),index=y.index))

0.00037646845237094683

In [ ]:
from sklearn.linear_model import ElasticNet

enet = ElasticNet(alpha=alpha, l1_ratio=0.7)

In [ ]:
from sklearn.svm import SVR

clf = SVR(C=1.0, epsilon=0.2)


In [ ]:
from sklearn.svm import NuSVR
clf = NuSVR(C=1.0, nu=0.1)


In [ ]:
from sklearn.linear_model import SGDRegressor()

clf = SGDRegressor()


In [ ]:
from sklearn.naive_bayes import GaussianNB
gnb = GaussianNB()

In [ ]:
from sklearn.ensemble import AdaBoostClassifier
clf = AdaBoostClassifier(n_estimators=100)